# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../synergy/gcc43.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- synergy-1.3.1/lib/platform/CXWindowsScreen.cpp.vanilla	2008-08-09 20:18:15.000000000 +0200
+++ synergy-1.3.1/lib/platform/CXWindowsScreen.cpp	2008-08-09 20:18:24.000000000 +0200
@@ -27,6 +27,8 @@
 #include "IEventQueue.h"
 #include "TMethodEventJob.h"
 #include <cstring>
+#include <stdlib.h>
+
 #if X_DISPLAY_MISSING
 #	error X11 is required to build synergy
 #else
--- synergy-1.3.1/lib/synergy/IKeyState.cpp.vanilla	2008-08-09 20:17:31.000000000 +0200
+++ synergy-1.3.1/lib/synergy/IKeyState.cpp	2008-08-09 20:17:38.000000000 +0200
@@ -14,6 +14,7 @@
 
 #include "IKeyState.h"
 #include <string.h>
+#include <stdlib.h>
 
 //
 // IKeyState
--- synergy-1.3.1/lib/synergy/IPrimaryScreen.cpp.vanilla	2008-08-09 20:17:49.000000000 +0200
+++ synergy-1.3.1/lib/synergy/IPrimaryScreen.cpp	2008-08-09 20:17:56.000000000 +0200
@@ -13,6 +13,7 @@
  */
 
 #include "IPrimaryScreen.h"
+#include <stdlib.h>
 
 //
 // IPrimaryScreen
--- synergy-1.3.1/lib/synergy/CPacketStreamFilter.cpp.vanilla	2008-08-09 20:17:08.000000000 +0200
+++ synergy-1.3.1/lib/synergy/CPacketStreamFilter.cpp	2008-08-09 20:17:17.000000000 +0200
@@ -12,6 +12,8 @@
  * GNU General Public License for more details.
  */
 
+#include <string.h>
+
 #include "CPacketStreamFilter.h"
 #include "IEventQueue.h"
 #include "CLock.h"
--- synergy-1.3.1/lib/net/CTCPSocket.cpp.vanilla	2008-08-09 20:15:30.000000000 +0200
+++ synergy-1.3.1/lib/net/CTCPSocket.cpp	2008-08-09 20:16:02.000000000 +0200
@@ -24,6 +24,7 @@
 #include "CArch.h"
 #include "XArch.h"
 #include <string.h>
+#include <stdlib.h>
 
 //
 // CTCPSocket
--- synergy-1.3.1/lib/base/CStringUtil.cpp.vanilla	2008-08-09 20:15:03.000000000 +0200
+++ synergy-1.3.1/lib/base/CStringUtil.cpp	2008-08-09 20:15:14.000000000 +0200
@@ -16,6 +16,7 @@
 #include "CArch.h"
 #include "common.h"
 #include "stdvector.h"
+#include <string.h>
 #include <cctype>
 #include <cstdio>
 #include <cstdlib>
--- synergy-1.3.1/lib/arch/CMultibyte.cpp.vanilla	2008-08-09 20:13:01.000000000 +0200
+++ synergy-1.3.1/lib/arch/CMultibyte.cpp	2008-08-09 20:13:35.000000000 +0200
@@ -24,8 +24,6 @@
 #endif
 #if HAVE_WCHAR_H || defined(_MSC_VER)
 #	include <wchar.h>
-#elif __APPLE__
-	// wtf?  Darwin puts mbtowc() et al. in stdlib
 #	include <stdlib.h>
 #else
 	// platform apparently has no wchar_t support.  provide dummy
--- synergy-1.3.1/lib/arch/CArchDaemonUnix.cpp.vanilla	2008-08-09 20:13:50.000000000 +0200
+++ synergy-1.3.1/lib/arch/CArchDaemonUnix.cpp	2008-08-09 20:14:48.000000000 +0200
@@ -15,6 +15,7 @@
 #include "CArchDaemonUnix.h"
 #include "XArchUnix.h"
 #include <unistd.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
--- synergy-1.3.1/lib/server/CClientProxy1_3.cpp.vanilla	2008-08-09 20:20:10.000000000 +0200
+++ synergy-1.3.1/lib/server/CClientProxy1_3.cpp	2008-08-09 20:20:20.000000000 +0200
@@ -18,6 +18,8 @@
 #include "IEventQueue.h"
 #include "TMethodEventJob.h"
 
+#include <string.h>
+
 //
 // CClientProxy1_3
 //
--- synergy-1.3.1/lib/server/CServer.cpp.vanilla	2008-08-09 20:20:30.000000000 +0200
+++ synergy-1.3.1/lib/server/CServer.cpp	2008-08-09 20:20:37.000000000 +0200
@@ -29,6 +29,7 @@
 #include "TMethodEventJob.h"
 #include "CArch.h"
 #include <string.h>
+#include <stdlib.h>
 
 //
 // CServer
--- synergy-1.3.1/lib/client/CClient.cpp.vanilla	2008-08-09 20:18:50.000000000 +0200
+++ synergy-1.3.1/lib/client/CClient.cpp	2008-08-09 20:19:17.000000000 +0200
@@ -27,6 +27,9 @@
 #include "IEventQueue.h"
 #include "TMethodEventJob.h"
 
+#include <string.h>
+#include <stdlib.h>
+
 //
 // CClient
 //
--- synergy-1.3.1/lib/client/CServerProxy.cpp.vanilla	2008-08-09 20:19:26.000000000 +0200
+++ synergy-1.3.1/lib/client/CServerProxy.cpp	2008-08-09 20:19:49.000000000 +0200
@@ -24,6 +24,7 @@
 #include "TMethodEventJob.h"
 #include "XBase.h"
 #include <memory>
+#include <string.h>
 
 //
 // CServerProxy
